bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -156,6 +159,10 @@ func (r *EKSConfigReconciler) joinWorker(ctx context.Context, log logr.Logger, c
 		}
 	}
 
+	if cluster.Spec.ControlPlaneRef == nil || cluster.Spec.ControlPlaneRef.Kind != "AWSManagedControlPlane" {
+		return ctrl.Result{}, errors.New("Cluster's controlPlaneRef needs to be an AWSManagedControlPlane in order to use the EKS bootstrap provider")
```

Comment:
This error could be a static package level error to help with `goerr113`. But thats not essential for this.